### PR TITLE
[CU-3rxr068] remove unnecessary .expect and add change clippy config

### PR DESCRIPTION
**Pull request by andresmechali**
## Issue

https://app.clickup.com/t/3rxr068

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-3rxr068 Replace every .expect on the Vesting pallet by proper errors](https://app.clickup.com/t/3rxr068)

File: code/parachain/frame/vesting/src/lib.rs [hunk 1]
```diff
@@ -606,7 +610,7 @@ impl<T: Config> Pallet<T> {
 
 					claims_per_schedule
 						.try_insert(*id_to_claim, available_amount)
```
Comment (vimukthi-git) [inline code/parachain/frame/vesting/src/lib.rs:612]: This `expect` is ok IMO as there is no chance(invariant) of it having a `MaxVestingSchedulesExceeded` given it is reinserting a value that was read in this same code.

File: code/parachain/frame/vesting/src/lib.rs [hunk 2]
```diff
@@ -34,9 +34,9 @@
 		clippy::disallowed_types,
 		clippy::indexing_slicing,
 		clippy::todo,
-		clippy::unwrap_used,
-		clippy::panic
-	)
+		clippy::expect_used
+	),
+	deny(clippy::unwrap_used, clippy::panic)
```
Comment (vimukthi-git) [inline code/parachain/frame/vesting/src/lib.rs:39]: What's the expected effect here?

Comment (andresmechali) [reply code/parachain/frame/vesting/src/lib.rs:39]: To forbid `unwrap` and `panic` and warn `expect`s.

File: code/parachain/frame/vesting/src/lib.rs (hunk 1 above)
Comment (andresmechali) [reply code/parachain/frame/vesting/src/lib.rs:612]: Yeah, but I was suggested not to have any `expect` besides the `GenesisBuild`, and I guess it doesn't hurt to remove it right?

**Comment by benluelo** [comment]
Why not make GenesisConfig.vesting a BoundedVec with the same size limit as the max vesting schedules?

File: code/parachain/frame/vesting/src/lib.rs (hunk 2 above)
Comment (vimukthi-git) [reply code/parachain/frame/vesting/src/lib.rs:39]: In the build we don't allow "warn" level stuff also, @benluelo is this line needed?

Comment (andresmechali) [reply code/parachain/frame/vesting/src/lib.rs:39]: Well I'll better remove those warnings as things like `#[pallet::pallet]` and the `expect`s from the `GenesisBuild` will trigger them.

Comment (benluelo) [reply code/parachain/frame/vesting/src/lib.rs:38]: `expect` should be allowed, and all uses of it should be audited

**Comment by andresmechali** [comment]
> Why not make GenesisConfig.vesting a BoundedVec with the same size limit as the max vesting schedules?

@benluelo that makes sense

**Comment by vimukthi-git** [comment]
> Why not make GenesisConfig.vesting a BoundedVec with the same size limit as the max vesting schedules?

This is ok, but I don't see this as a critical fix for release as the genesis is already done. So let's keep that separate.

**Comment by andresmechali** [comment]
> > Why not make GenesisConfig.vesting a BoundedVec with the same size limit as the max vesting schedules?
> 
> This is ok, but I don't see this as a critical fix for release as the genesis is already done. So let's keep that separate.

Alright. Do we even need to handle schedules in the `GenesisBuild`? (see https://app.clickup.com/t/3rxr068?comment=1795517562)

**Pull request closed without merging**